File: source/icdab_wx/icdab_pt/ViewController.m
//
//  ViewController.m
//  icdab_pt
//
//  Created by Faisal Memon on 09/10/2020.
//

#import "ViewController.h"

#import "Memory.h"

@interface ViewController ()

@end

@implementation ViewController

- (IBAction)triggerMemmapCrashAction:(id)sender {
    [Memory map_jit_memoryStalled:NO];
}

- (IBAction)triggerStalledCrashAction:(id)sender {
    [Memory map_jit_memoryStalled:YES];
}

- (void)viewDidLoad {
    [super viewDidLoad];
}


@end

File: source/icdab_diag/icdab_diag/ViewController.swift
//
//  ViewController.swift
//  icdab_diag
//
//  Created by Faisal Memon on 20/10/2020.
//

import UIKit
import MetricKit

class ViewController: UIViewController {

    let memory = Memory()
    let metrics = MXMetricManager.shared
    
    override func viewDidLoad() {
        super.viewDidLoad()
        metrics.add(self)
        for item in metrics.pastDiagnosticPayloads {
            print(item)
        }
        for item in metrics.pastPayloads {
            print(item)
        }
    }

    @IBAction func dereferenceBadPointerButtonAction(_ sender: UIButton) {
        memory.dereferenceBadPointer()
    }
    
}

extension ViewController: MXMetricManagerSubscriber {
    func didReceive(_ payloads: [MXMetricPayload]) {
        for payload in payloads {
            print(payload)
        }
    }
    
    func didReceive(_ payloads: [MXDiagnosticPayload]) {
        for payload in payloads {
            print(payload)
        }
    }
}

